pkg/operator/secretannotator/azure/status.go

```diff
@@ -0,0 +1,30 @@
+package azure
+
+import (
+	log "github.com/sirupsen/logrus"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/secretannotator/status"
+	"github.com/openshift/cloud-credential-operator/pkg/util/clusteroperator"
+)
+
+// syncOperatorStatus computes the operator's current status and
+// creates or updates the ClusterOperator resource for the operator accordingly.
+func (r *ReconcileCloudCredSecret) syncOperatorStatus() error {
+	return status.SyncOperatorStatus(r.Client)
+}
+
+var _ clusteroperator.StatusHandler = &ReconcileCloudCredSecret{}
```

Comment:
Is there a reason why we need the reconciler itself to be a `StatusHandler`? Could we define a type in pkg/operator/secretannotator/status/status.go that was the `StatusHandler` instead? That way we do not need this boilerplate code for each platform.

Reply:
excellent suggestion. done.